src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -670,19 +646,23 @@ private void TakeIncrementalSnapshot(BoundNode node)
 
         private void SetUpdatedSymbol(BoundNode node, Symbol originalSymbol, Symbol updatedSymbol)
         {
-            if (_updatedSymbolMapOpt is object)
+#if DEBUG
+            Debug.Assert(node is object);
+            Debug.Assert(AreCloseEnough(originalSymbol, updatedSymbol), $"Attempting to set {node.Syntax} from {originalSymbol.ToDisplayString()} to {updatedSymbol.ToDisplayString()}");
+#endif
+
+            if (_snapshotBuilderOpt is object)
             {
-                var key = (node, originalSymbol);
-                if ((object)originalSymbol != updatedSymbol)
+                if (!Symbol.Equals(originalSymbol, updatedSymbol, TypeCompareKind.ConsiderEverything))
                 {
-                    _updatedSymbolMapOpt[key] = updatedSymbol;
+                    _snapshotBuilderOpt.SetUpdatedSymbol(node, originalSymbol, updatedSymbol);
```

Comment:
>_snapshotBuilderOpt.SetUpdatedSymbol(node, originalSymbol, updatedSymbol); [](start = 20, length = 74)

It looks like we will always get here for a lambda and will never get into else for a lambda. Is this expected?

Reply:
No, will fix.

---
In reply to: [329723903](https://github.com/dotnet/roslyn/pull/38769#discussion_r329723903) [](ancestors = 329723903)